src/hotspot/share/memory/arena.cpp

```diff
@@ -43,41 +44,32 @@ STATIC_ASSERT(is_aligned((int)Chunk::medium_size, ARENA_AMALLOC_ALIGNMENT));
 STATIC_ASSERT(is_aligned((int)Chunk::size, ARENA_AMALLOC_ALIGNMENT));
 STATIC_ASSERT(is_aligned((int)Chunk::non_pool_size, ARENA_AMALLOC_ALIGNMENT));
 
-//--------------------------------------------------------------------------------------
-// ChunkPool implementation
-
 // MT-safe pool of same-sized chunks to reduce malloc/free thrashing
 // NB: not using Mutex because pools are used before Threads are initialized
 class ChunkPool {
-  Chunk*       _first;        // first cached Chunk; its first word points to next chunk
-  const size_t _size;         // (inner payload) size of the chunks this pool serves
-
   // Our four static pools
   static const int _num_pools = 4;
   static ChunkPool _pools[_num_pools];
 
- public:
-  ChunkPool(size_t size) : _first(nullptr), _size(size) {}
+  Chunk*       _first;
+  const size_t _size;         // (inner payload) size of the chunks this pool serves
 
-  // Allocate a chunk from the pool; returns null if pool is empty.
-  Chunk* allocate() {
+  // Returns null if pool is empty.
+  Chunk* take_from_pool() {
     ThreadCritical tc;
     Chunk* c = _first;
     if (_first != nullptr) {
       _first = _first->next();
     }
     return c;
   }
-
-  // Return a chunk to the pool
-  void free(Chunk* chunk) {
+  void return_to_pool(Chunk* chunk) {
     assert(chunk->length() == _size, "wrong pool for this chunk");
     ThreadCritical tc;
     chunk->set_next(_first);
     _first = chunk;
   }
 
-  // Prune the pool
   void prune() {
```

Comment:
Admittedly the old comment was useless, but could you add a comment explaining what happens?

Alternatively, rename the function. E.g. free_all_chunks or similar. It is not really pruning anything since it completely empties out the list (pruning suggests there is something pruned left afterwards).